assignments/prototype-refactor.js

```diff
@@ -7,3 +7,146 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+class GameObject {
+  constructor(dimensions) {
+    this.createdAt = new Date();
+    this.dimensions = dimensions.dimensions;
+  }
+  destroy() {
+    return `${this.name} was removed from the game.`;
+  }
+}
+
+class CharacterStats extends GameObject {
+  constructor(characterAttributes) {
+    super(characterAttributes);
+    this.hp = characterAttributes.hp;
+    this.name = characterAttributes.name;
+  }
+  takeDamage() {
+    return `${this.name} took damage.`;
+  }
+}
+
+class Humanoid extends CharacterStats {
+  constructor(humanTraits) {
+    super(humanTraits);
+    this.faction = humanTraits.faction;
+    this.weapons = humanTraits.weapons;
+    this.language = humanTraits.language;
+  }
+  greet() {
+    return `${this.name} offers a greeting in ${this.language}.`
+  }
+}
+
+class Villian extends Humanoid {
+  constructor(villianTraits) {
+    super(villianTraits);
+    this.blackMagic = function () {
```

Comment:
You're the first person I've seen declaring your function in the constructor. Good job. Make the Villain and Hero constructors look very clean in this case.